src/cookie-io.js

```diff
@@ -0,0 +1,226 @@
+import { defaultData } from "./default-data.js";
+import { CONSTANTS } from "./constants/constants.js";
+
+function getDefaultData() {
+    // Reset data (deep copy)
+    return JSON.parse(JSON.stringify(defaultData));
+}
+
+class CharacterData {
+    characterData = getDefaultData();
+
+    getName() {
+        return this.characterData.name;
+    }
+    setName(name) {
+        this.characterData.name = name;
+    }
+
+    addGold(amount) {
+        if (this.characterData.gold + amount >= 0) {
+            this.characterData.gold += amount;
+        } else {
+            console.log(
```

Comment:
console log bad

Reply:
I'm going to make the chat window work as a functioning log later, so it's a good placeholder